src/cloudWatchLogs/commands/viewLogStream.ts

```diff
@@ -35,7 +35,9 @@ export async function viewLogStream(node: LogGroupNode, registry: LogStreamRegis
             logStreamResponse.logStreamName,
             logStreamResponse.region
         )
-        await registry.upsertLog(uri)
+        if (!registry.hasLog(uri)) {
+            await registry.upsertLog(uri)
```

Comment:
isn't the purpose of "upsert" to handle the case where `hasLog()` is true? So it's a bit weird to have an explicit check for `!hasLog()` before calling upsert.

Reply:
Yeah, I originally had this as a separate create and update, will probably move back to that now that I think about it. The fact that I moved to this simplified version was an artifact of the editor not updating.